retrieval_service/datastore/providers/alloydb.py

```diff
@@ -532,90 +172,30 @@ async def insert_ticket(
-                    :flight_number,
-                    :departure_airport,
-                    :arrival_airport,
-                    :departure_time,
-                    :arrival_time
-                );
-            """
-            )
-            params = {
-                "user_id": user_id,
-                "user_name": user_name,
-                "user_email": user_email,
-                "airline": airline,
-                "flight_number": flight_number,
-                "departure_airport": departure_airport,
-                "arrival_airport": arrival_airport,
-                "departure_time": departure_time_datetime,
-                "arrival_time": arrival_time_datetime,
-            }
-            result = (await conn.execute(s, params)).mappings()
-            await conn.commit()
-            if not result:
-                raise Exception("Ticket Insertion failure")
+        return await self.__pg_ds.insert_ticket(
+            user_id,
+            user_name,
+            user_email,
+            airline,
+            flight_number,
+            departure_airport,
+            arrival_airport,
+            departure_time,
+            arrival_time,
+        )
 
     async def list_tickets(
         self,
         user_id: str,
     ) -> tuple[list[Any], Optional[str]]:
-        async with self.__pool.connect() as conn:
-            sql = """
-                SELECT user_name, airline, flight_number, departure_airport, arrival_airport, departure_time, arrival_time FROM tickets
-                WHERE user_id = :user_id
-                """
-            s = text(sql)
-            params = {
-                "user_id": user_id,
-            }
-            results = (await conn.execute(s, params)).mappings().fetchall()
-
-        res = [r for r in results]
-        return res, sql
+        return await self.__pg_ds.list_tickets(user_id)
 
     async def policies_search(
         self, query_embedding: list[float], similarity_threshold: float, top_k: int
     ) -> tuple[list[str], Optional[str]]:
-        async with self.__pool.connect() as conn:
-            sql = """
-                SELECT content
-                FROM policies
-                WHERE (embedding <=> :query_embedding) < :similarity_threshold
-                ORDER BY (embedding <=> :query_embedding)
-                LIMIT :top_k
-                """
-            s = text(sql)
-            params = {
-                "query_embedding": query_embedding,
-                "similarity_threshold": similarity_threshold,
-                "top_k": top_k,
-            }
-            results = (await conn.execute(s, params)).mappings().fetchall()
-
-        res = [r["content"] for r in results]
-        return res, sql
+        return await self.__pg_ds.policies_search(
+            query_embedding, similarity_threshold, top_k
+        )
 
     async def close(self):
-        await self.__pool.dispose()
+        return await self.__pg_ds.close()
```

Comment:
Is return needed here?

Reply:
You're right, the potential confusion outweighs the future benefit and consistency with the methods above.  I'll remove the `return` in a separate PR. Thanks!